SEGGER/SEGGER_RTT.c

```diff
@@ -1902,7 +1925,22 @@ int SEGGER_RTT_SetFlagsDownBuffer(unsigned BufferIndex, unsigned Flags) {
 *
 */
 void SEGGER_RTT_Init (void) {
+  /*
+   * The standard Segger code calls only _DoInit() here. This happens,
+   * for example, in a bootloader or in stand-alone applications.
+   * For applications started by a bootloader, Segger recommends not
+   * calling this function, as all the recommended APIs use INIT().
+   * Unfortunately Zephyr log backend uses SEGGER_RTT_WriteSkipNoLock()
+   * that doesn't call INIT(), so the additional two following variants
+   * have been added here for a safe API usage.
+   */
+#if defined(CONFIG_SEGGER_RTT_INIT_MODE_STRONG_CHECK)
+  SEGGER_RTT_StrongCheckInit();
+#elif defined(CONFIG_SEGGER_RTT_INIT_MODE_WEAK_CHECK)

```

Comment:
Where is `CONFIG_SEGGER_RTT_INIT_MODE_WEAK_CHECK` declared?

Reply:
I expected them to come from here https://github.com/zephyrproject-rtos/zephyr/pull/53569/files#diff-fdfeddb6c80762ec457334a8b543499a5d6dc18ddf20b82a6f81d70d08c56670

Reply:
same as above